File: examples/EnableDigitalAgentsDemo.java
package com.cisco.ccbu.cce.unifiedconfig.toolkit.examples;


import com.cisco.ccbu.cce.unifiedconfig.toolkit.RESTClient;
import com.cisco.ccbu.cce.unifiedconfig.toolkit.bean.Agent;


/**
 * Example code for creating a digital agent.
 *
 * Command line launch: mvn exec:java -Dexec.mainClass="com.cisco.ccbu.cce.unifiedconfig.toolkit.examples.EnableDigitalAgentsDemo" -Dexec.args="10.86.135.193 administrator@test.com abc123="
 */
public class EnableDigitalAgentsDemo {
    private static RESTClient restClient = null;

    public static void main(String[] args) {

        if(args.length < 3){
            System.out.println("EnableDigitalAgentsDemo requires 3 parameters: hostname username (with @domain) password");
            System.exit(0);
        }

        // Create a new RESTClient object with the IP of AW HDS
        restClient = new RESTClient(args[0], args[1], args[2]);

        try {
            enableDigitalAgentsDemo();
        } catch (Exception e){
            System.out.println("Exception occured: " + e.getMessage());
        }
    }

    private static void enableDigitalAgentsDemo() throws ClassNotFoundException {
        String payload = "{\"person\": {" +
                "        \"digitalChannel\": true," +
                "        \"emailAddress\": \"xyz@cisco.com\"," +
                "        \"ecePerson\": false," +
                "        \"firstName\": \""+DemoUtils.generateUniqueString("firstName", 1)+"\"," +
                "        \"lastName\": \""+DemoUtils.generateUniqueString("lastName", 1)+"\"," +
                "        \"loginEnabled\": true," +
                "        \"ssoEnabled\": true," +
                "        \"userName\": \""+DemoUtils.generateUniqueString("userName", 1)+"\"" +
                "    }" +
                "}";

        // -- Create and get the agent
        Agent newAgent = restClient.createAndGet_JSONPayload(payload, "agent", Class.forName("com.cisco.ccbu.cce.unifiedconfig.toolkit.bean.Agent"));
        System.out.println("Digital Agent created: " + newAgent.getRefURL());
        System.out.println("Digital Agent Email ID: " + newAgent.getPerson().getEmailAddress());

        // delete the digital agent
        restClient.delete(newAgent.getRefURL());
        System.out.println("Deleted Digital Agent: " + newAgent.getRefURL());
    }
}

File: PcceExamples/EnableDigitalAgentsDemo.cs
using CiscoPCCE.Toolkit.Bean;
using CiscoPCCE.Toolkit.Sdk;

namespace CiscoPCCE.Toolkit.Examples
{
    /// <summary>
    /// Example code for creating a digital agent.
    /// </summary>
    public class EnableDigitalAgentsDemo
    {
        private static CiscoRestClient? restClient = null;

        public static async Task Main(string[] args)
        {
            if (args.Length < 3)
            {
                Console.WriteLine("EnableDigitalAgentsDemo requires 3 parameters: hostname username (with @domain) password");
                Environment.Exit(0);
            }

            // Create a new RESTClient object with the IP of AW HDS
            restClient = new CiscoRestClient(args[0], args[1], args[2]);

            try
            {
                await EnableDigitalAgentsDemoAsync();
            }
            catch (Exception e)
            {
                Console.WriteLine($"Exception occurred: {e.Message}");
            }
        }

        private static async Task EnableDigitalAgentsDemoAsync()
        {
            // Create a digital agent using the Agent class
            var newAgent = new Agent
            {
                Person = new Person
                {
                    DigitalChannel = true,
                    EmailAddress = "xyz@cisco.com",
                    EcePerson = false,
                    FirstName = DemoUtils.GenerateUniqueString("firstName", 1),
                    LastName = DemoUtils.GenerateUniqueString("lastName", 1),
                    LoginEnabled = true,
                    SsoEnabled = true,
                    UserName = DemoUtils.GenerateUniqueString("userName", 1),
                    Department = new ReferenceBean()
                },
                AgentDeskSettings = new ReferenceBean(),
                AgentTeam = new ReferenceBean(),
                Datacenter = new ReferenceBean(),
                DefaultSkillGroup = new ReferenceBean(),
                Department = new ReferenceBean(),
                Peripheral = new PeripheralRef(),
                PeripheralSet = new ReferenceBean()
            };

            // -- Create and get the agent
            var refUrl = await restClient!.CreateAndGetAsync(newAgent);
            Agent? createdAgent = null;
            if (refUrl != null)
            {
                createdAgent = await restClient.GetAsync<Agent>(refUrl);
                Console.WriteLine($"Digital Agent created: {createdAgent?.RefURL}");
                Console.WriteLine($"Digital Agent Email ID: {createdAgent?.Person?.EmailAddress}");
            }

            // delete the digital agent
            if (createdAgent?.RefURL != null)
            {
                await restClient.DeleteAsync(createdAgent.RefURL);
                Console.WriteLine($"Deleted Digital Agent: {createdAgent.RefURL}");
            }
        }
    }
}
